core-common/src/main/java/org/zalando/nakadi/domain/NakadiRecord.java

```diff
@@ -82,7 +83,7 @@ public static NakadiRecord fromAvro(final String eventTypeName,
                 }));
         return new NakadiRecord(
                 eventTypeName,
-                null,
+                partition,
```

Comment:
can take it from `metadata` then it won't be possible to mix up partition as a param and partition from metadata